src/kpi/statistics/couplingOfComponents/couplingOfComponents.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { RepositoryService } from 'src/entities/repositories/repository.service';
+import { RepositoryIdentifier } from 'src/entities/repositories/model';
+import { getPrFilesQuery } from './lib/prFilesQuery';
+import { getCoupling } from './lib/prFilesUtil';
+import { coupling } from './model/coupling';
+
+@Injectable()
+export class CouplingOfComponents {
+  private readonly logger = new Logger(CouplingOfComponents.name);
+
+  constructor(private readonly repoService: RepositoryService) {}
+
+  /**
+   * Computes KPI Coupling Of Compontents for @param repoIdent.
+   * It exludes files from @param fileFilter and limits number of diffs
+   * to @param diffsLimit, it applies size of coupling @param couplingSize
+   * the amount of @param occs. Bounds to interval @param since until @param to.
+   * @returns all couplings sorted descending.
+   */
+  async couplingOfComponents(
+    repoIdent: RepositoryIdentifier,
+    diffsLimit?: number,
+    fileFilter?: string[],
+    couplingSize?: number,
+    occs?: number,
+    since: string = undefined,
+    to: string = undefined,
+  ) {
+    const lookUpQuery = this.repoService.preAggregate(repoIdent, {
+      diffs: { prFiles: true, since: since, to: to },
+    });
+
+    const prFilesQuery = getPrFilesQuery(lookUpQuery, diffsLimit, fileFilter);
+
+    const prFiles = await prFilesQuery.exec();
+    console.log(prFiles);
```

Comment:
use the logger you created in line 10 !!!!

Reply:
Agreed

Reply:
I am very sorry ... I forgot to delete this. It does not work to display it in that way with logger. Anyway, I got Postman :)